do tutorial suggestions (#923)
n8n-io/n8n-docs: docs/hosting/server-setups/digital-ocean.md

# Hosting n8n on DigitalOcean

This hosting guide shows you how to self-host n8n on a DigitalOcean droplet. It uses:

* [Caddy](http://caddyserver.com){:target="_blank" .external-link} (a reverse proxy) to allow access to the Droplet from the internet. 
* [Docker Compose](https://docs.docker.com/compose/){:target="_blank" .external-link} to create and define the application components and how they work together.

> * The [n8n Docker image](https://hub.docker.com/r/n8nio/n8n){:target=_blank .external-link}.

Reviewer: Hmm. If you really want to mention this, then really you should probably also mention that it uses the Caddy one, but personally, I'm not sure if this is really necessary to say as using Docker would imply that you're using images for the services anyway. Maybe a compromise might be to say in the line above that the Docker compose file uses those images, or actually would make more sense to mention that when discussing the contents of the Docker compose file I think.

## Create a Droplet

1. [Log in](https://cloud.digitalocean.com/login){:target=_blank .external-link} to DigitalOcean. 
2. Select the project to host the Droplet, or [create a new project](https://docs.digitalocean.com/products/projects/how-to/create/){:target=_blank .external-link}.
3. In your project, select **Droplets** from the **Manage** menu. 
4. [Create a new Droplet](https://docs.digitalocean.com/products/droplets/how-to/create/){:target=_blank .external-link} using the [Docker image](https://marketplace.digitalocean.com/apps/docker){:target="_blank" .external-link} available on the **Marketplace** tab.

!!! note "Droplet resources"

> 		When creating the Droplet, DigitalOcean asks you to choose a plan. For most usage levels, a basic shared CPU plan is enough.

Reviewer: Is that true? I honestly don't know…

Author: Yes. I checked.

!!! note "SSH or Password"

> 		DigitalOcean allows you to choose between SSH and password-based authentication. SSH is more secure. The rest of this guide assumes you are using SSH.

Reviewer: ```suggestion
		DigitalOcean lets you to choose between SSH and password-based authentication. SSH is more secure. The rest of this guide assumes you are using SSH.
```

## Log in to your Droplet

> The rest of this guide requires you to log in to the Droplet using a terminal with SSH. Refer to [How to Connect to Droplets with SSH](https://docs.digitalocean.com/products/droplets/how-to/connect-with-ssh/){:target="_blank" .external-link} for guidance.

Reviewer: ```suggestion
The rest of this guide requires you to log in to the Droplet using a terminal with SSH. Reado [How to Connect to Droplets with SSH](https://docs.digitalocean.com/products/droplets/how-to/connect-with-ssh/){:target="_blank" .external-link} for more information.
```

Personally… Sounds a bit friendlier.

## Create folders and files

Both n8n and Caddy require creating folders that the host operating system (the DigitalOcean Droplet) copies to Docker containers to make them available to Docker.

> Create the following in the Droplet in your home folder:

Reviewer: Not sure this is correct… I always thought it was "on", but I also can't seem to find anything definitive… It's a server, and typically you say "on a server", but I think I was more trying to avoid repeated "on"s.

Maybe…

```suggestion
Create the following in the Droplet root user's home folder:
```

Which is actually also more accurate now that I think about it.

### Create Docker volume

To persist the Caddy cache between restarts and speed up start times, create [a Docker volume](https://docs.docker.com/storage/volumes/){:target="_blank" .external-link} that Docker reuses between restarts:

```shell
docker volume create caddy_data
```

## Set up DNS

> n8n typically operates on a subdomain. Create a DNS record with your provider for the subdomain and point it to the IP address of the Droplet. The exact steps for this depend on your DNS provider, but typically you need to create a new A record for the n8n subdomain. DigitalOcean provide [An Introduction to DNS Terminology, Components, and Concepts](https://www.digitalocean.com/community/tutorials/an-introduction-to-dns-terminology-components-and-concepts){:target="_blank" .external-link}.

Reviewer: ```suggestion
n8n typically operates on a subdomain. Create a DNS record with your provider for the subdomain and point it to the IP address of the Droplet. The exact steps for this depend on your DNS provider, but typically you need to create a new "A" record for the n8n subdomain. DigitalOcean provide [An Introduction to DNS Terminology, Components, and Concepts](https://www.digitalocean.com/community/tutorials/an-introduction-to-dns-terminology-components-and-concepts){:target="_blank" .external-link}.
```

or whatever formatting you want to use, just so it doesn't look like a typo 😆

SSL_EMAIL=example@example.com
```

> For more information on n8n environment variables, refer to [Environment variables](/hosting/environment-variables/).

Reviewer: ```suggestion
For more information on n8n environment variables, read the [Environment variables](/hosting/environment-variables/) documentation.
```

This is persona, just how I have always constructed reference links, but if you have your own style, go for it.

## Configure Caddy

Caddy needs to know which domains it should serve, and which port to expose to the outside world. Create a `Caddyfile` file in the `caddy_config` folder.

```shell
nano caddy_config/Caddyfile
```

> Add the following configuration, adding your domain. If you followed the steps to name the subdomain n8n, your full domain will be similar to `n8n.example.com`. The `n8n` in the `reverse_proxy` setting tells Caddy to use the service definition defined in the `docker-compose.yml` file:

Reviewer: ```suggestion
Add the following configuration, adding your domain. If you followed the steps to name the subdomain n8n, your full domain is similar to `n8n.example.com`. The `n8n` in the `reverse_proxy` setting tells Caddy to use the service definition defined in the `docker-compose.yml` file:
```
